specification/web/resource-manager/Microsoft.DomainRegistration/stable/2022-09-01/Domains.json

```diff
@@ -0,0 +1,1702 @@
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "azureResourceName": {
+          "description": "Name of the Azure resource the hostname is assigned to. If it is assigned to a Traffic Manager then it will be the Traffic Manager name otherwise it will be the app name.",
+          "type": "string"
+        },
+        "azureResourceType": {
+          "description": "Type of the Azure resource the hostname is assigned to.",
+          "enum": [
+            "Website",
+            "TrafficManager"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "AzureResourceType",
+            "modelAsString": false
+          }
+        },
+        "customHostNameDnsRecordType": {
+          "description": "Type of the DNS record.",
+          "enum": [
+            "CName",
+            "A"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "CustomHostNameDnsRecordType",
+            "modelAsString": false
+          }
+        },
+        "hostNameType": {
+          "description": "Type of the hostname.",
+          "enum": [
+            "Verified",
+            "Managed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "HostNameType",
+            "modelAsString": false
+          }
+        }
+      }
+    },
+    "NameIdentifierCollection": {
+      "description": "Collection of domain name identifiers.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "../../../Microsoft.Web/stable/2022-09-01/CommonDefinitions.json#/definitions/NameIdentifier"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "subscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "description": "Your Azure subscription ID. This is a GUID-formatted string (e.g. 00000000-0000-0000-0000-000000000000).",
+      "required": true,
+      "type": "string"
+    },
+    "resourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "description": "Name of the resource group to which the resource belongs.",
+      "required": true,
+      "type": "string",
+      "maxLength": 90,
+      "minLength": 1,
+      "pattern": "^[-\\w\\._\\(\\)]+[^\\.]$",
+      "x-ms-parameter-location": "method"
+    },
+    "apiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "description": "API Version",
+      "required": true,
+      "type": "string"
+    }
```

Comment:
There are definitions of common parameters in commontypes types.json that you can use by $ref instead of declaring your own, like SubscriptionIdParameter, OperationIdParameter, and ApiVersionParameter,
See
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630)